src/python/Utils/procstats.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+#-*- coding: utf-8 -*-
+#pylint: disable=
+"""
+File       : procstats.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: This module provide ability to get detailed information about
+certain UNIX process, referred by PID. It is based on psutil and threading
+python modules. It can be used in any web/data-server to get live status
+about the running server. Here is an example
+
+from Utils.procstats import status thread_stack
+
+
+class SomeServer(object):
+    def __init__(self):
+        self.time0 = time.time()
+    def status(self):
+	sdict = {'server': status()}
+        sdict['server'].update({'uptime': time.time()-self.time0})
+        sdict.update(thread_stack())
+	return sdict
+
+The exposed status endpoint will return a json dictionary containing server
+upteima and cpu/mem/threads information.
+"""
+
+# system modules
+import os
+import sys
+import json
+import time
+import argparse
+import threading
+import traceback
+
+try:
+    import psutil
+except:
+    pass
+
+def timestamp(time0=None):
+    "Standard timestamp"
+    ttime = time.localtime(time0) if time0 else time.localtime()
+    tstamp = time.strftime('%d/%b/%Y:%H:%M:%S', ttime)
+    return tstamp
+
+def status(pid=None):
+    "Return status of a process"
+    if  not pid:
+        pid = os.getpid()
+    pdict = {'pid': pid, 'timestamp': timestamp(), 'time':time.time()}
+    if  not 'psutil' in sys.modules:
+        return pdict
+    proc = psutil.Process(pid)
+    pdict.update(proc.as_dict())
+    return pdict
+
+def thread_stack():
+    """
+    Dump context of all threads upon given signal
+    http://stackoverflow.com/questions/132058/showing-the-stack-trace-from-a-running-python-application
+    """
+    tdict = {}
+    id2name = dict([(th.ident, th.name) for th in threading.enumerate()])
+    threads = []
+    for tid, stack in sys._current_frames().items():
+        tdict = {"thead":id2name.get(tid,""), "thead_id": tid}
+        stack_list = []
+        for filename, lineno, name, line in traceback.extract_stack(stack):
+            sdict = dict(filename=filename, line_number=lineno, name=name, line=line)
+            stack_list.append(sdict)
+        tdict.update({"stack":stack_list})
+        threads.append(tdict)
+    return dict(threads=threads)
+
+class OptionParser():
```

Comment:
Since we're going to use it as a module, I believe this class and the `main` function are not really needed here, right?